packages/source-fetcher/lib/sourcify.ts

```diff
@@ -52,50 +52,62 @@ const SourcifyFetcher: FetcherConstructor = class SourcifyFetcher
   async fetchSourcesForAddress(
     address: string
   ): Promise<Types.SourceInfo | null> {
-    const metadata = await this.getMetadata(address);
+    let result = await this.fetchSourcesForAddressAndMatchType(address, "full");
+    if (!result) {
+      //if we got nothing when trying a full match, try for a partial match
+      result = await this.fetchSourcesForAddressAndMatchType(address, "partial");
+    }
+    //if partial match also fails, just return null
+    return result;
+  }
+
+  private async fetchSourcesForAddressAndMatchType(
+    address: string,
+    matchType: "full" | "partial"
+  ): Promise<Types.SourceInfo | null> {
+    const metadata = await this.getMetadata(address, matchType);
     debug("metadata: %O", metadata);
     if (!metadata) {
       debug("no metadata");
       return null;
     }
     let sources: Types.SourcesByPath;
-    if (metadata.language === "Solidity") {
-      sources = Object.assign(
-        {},
-        ...(await Promise.all(
-          Object.entries(metadata.sources).map(
-            async ([sourcePath, {content: source}]) => ({
-              [sourcePath]:
-                source !== undefined
-                  ? source //sourcify doesn't support this yet but they're planning it
-                  : await this.getSource(address, sourcePath)
-            })
-          )
-        ))
-      );
-    } else {
-      //don't bother attempting to fetch sources if it's Yul
-      sources = {};
-    }
+    sources = Object.assign(
```

Comment:
So what does the removal of the check for language mean? So the sources can be Solidity or Yul...can they be Vyper?

Reply:
This is explained under #4.  The language check is removed because it's unnecessary, because the sources can only ever be Solidity in the first place.  Yul and Vyper don't have metadata, so there's not some way to use them with Sourcify.